guidance/library/_json.py

```diff
@@ -297,19 +336,21 @@ def _gen_json(
             )
         raise ValueError(f"Unsupported type in schema: {target_type}")
 
-    raise ValueError(f"Can't process JSON node: {json_schema}")
```

Comment:
Shouldn't this still get raised _somewhere_?

Reply:
Good question. The fact that empty schemas imply arbitrary json makes it such that we can't use the _absence_ of keys to raise this exception. I expect that the schema validation we do also implies that we can't get _conflicting_ or _incompatible_ keys... So maybe all that's left is some assertion about extra/unhandled keys? 

E.g. we should raise errors if users ask to match a pattern, provide an integer range, etc. -- any other situation you can think of?

Reply:
One pitfall to be aware of here if you decide to implement more complex checking before falling back to `any` generation is that there are a few human-readable-only JSON schema elements: https://json-schema.org/understanding-json-schema/reference/annotations

So e.g. `{"description": "Any arbitrary JSON object"}` is considered an empty schema under the spec.

Reply:
I added some logic for validating keys of the json schema to ensure we only allow keys that are either on a whitelist of "implemented" or "ignored" (title, description, ...). An explicit blacklist of "things we don't support" feels like a far more fragile approach, hence me not taking it.

@riedgar-ms would you mind taking a look at the general approach to validation?

Note that tests are now failing because the validation raises an exception telling us that we don't support the `required` key that we used in a lot of our tests for object. Fixing that added to my TODOs. It's weirdly non-trivial to get the commas right when separating mixtures of optional and non-optional values...